event.go

```diff
@@ -23,4 +23,6 @@ type ScrollEvent struct {
 type DragEvent struct {
 	PointEvent
 	DraggedX, DraggedY int
+	DraggedTotal       Position
+	ElementStartPos    Position
```

Comment:
I'm a little concerned about new names here - which "Element" does this refer to, and what is "Start"?
If I am listening to the drag event then I assume that the element is me and that start is where I was when dragging began?

Isn't obj.Position().X + DraggedX the same as ElementStartPos.X + DraggedTotal.X?
Sorry if I'm missing something obvious but I want to avoid adding superfluous APIs to the events

Reply:
Yeah, I'm not that good in naming … getting names right is another advantage of PRs IMHO.

Let's say you have moved the element (a scroll bar for instance) over the bounds. The scroller would then stop to follow your dragging. If you decide to move back, you'll get a negative dragged distance which would move the scroller back even if the mouse is still far beyond the boundary.
That's why the dragged total is needed.

But the scroller can't use the dragged total as its start position because it might be that the dragging started in the middle of the scrollable area. Using the total dragged as only source of position would push it back to the top in this case.
That's why the start position is needed. That's the position the dragged element had at the start of this dragging.
`obj.Position().X + DraggedX` is _not_ the same as `ElementStartPos.X + DraggedTotal.X` because the object's position may change by a different amount than the dragged distance (if the scroller hits a boundary).


Conclusion:
* We need a better name for `ElementStartPos`. Your assumption regarding its meaning is totally correct. It is the position the dragged element had as the dragging started. Any ideas on a better name?
* There might be an alternative with less data to compute the needed numbers but I'm not aware of it.